### DISCUSSION TITLE
Low tech opinion question about auto-reload/actix replacing php

### DISCUSSION THREAD

**Discussion by surfingreg** [sunng87/handlebars-rust#528] [Q&A]
First of all, thank you for this Handlebars crate!

Scenario: I'm trying to jam a square rust peg (that I really love) into a round (more php-shaped) hole having inherited a small shop of biz-oriented programmers who've long maintained a php/crud site not originally built by them. My dream would be to replace the php code with Actix and Handlebars. The others don't totally care, but don't need to be bothered beyond the details of adding boxes and pages to the existing site. I can't ditch the whole site, but personally/secretly really really do not like php. I also can't completely re-tool to react (nor do I want to, we're staying server-side). So, we're a shop of devs who "build" by writing "code" and quickly looking to the browser to see if it "works". Yes. Okay. Anyway. Let's say those are the constraints. Could we replace the A and P in that lAmP stack with Actix and Handlebars? Without terribly traumatizing my team?

Options:
1.  ```cargo watch -x run``` They edit html, handlebars, some rust for some dynamically generated html, basically crud off the web forms they build. The usual php stuff. See results immediately in browser. Not sure I could get way with compile time waits. Maybe.
2. Perhaps obviously(?), dev_mode doesn't seem to do anything in my excerpted code below. Is there a simple fix to at least get it to work in Actix?
3. I've looked at binserve (from the Actix examples page). If all we did was static pages it'd work. But it's kind of the idea I guess.

This probably isn't a well thought out problem because eventually they'll need to compile Rust. For now I fully welcome suggestions to get dev_mode or hot-reload of templates at least under Actix to work. And beyond that any recommendations on how to get off php without going full client-side js. It seems Handlebars combine with Actix has real promise. I just need to bring the tech (we)/I love to the level of the team I have.

```
/// handlebars instance for actix to pass around 
fn start_handlebars() -> Data<Handlebars<'static>> {
    let mut handlebars = Handlebars::new();
    handlebars.register_templates_directory(".html", "./src/html").unwrap();
    handlebars.set_strict_mode(true);
    
    handlebars.dev_mode();
    Data::new(handlebars)
}

/// GET /
pub async fn home(hb: web::Data<Handlebars<'_>>) -> HttpResponse {
    let data = serde_json::json!({
            "title": page_title,
            "parent": "base_template",
    });
    let body = hb.render("home", &data).unwrap();
    HttpResponse::Ok().body(body)
}

...
// main excerpt where Actix starts...
    HttpServer::new(move || {
        App::new()
            .app_data(start_handlebars())
            .route("/", web::get().to(home))

```

**Comment by sunng87** [comment]
@gregparsons sorry for super late response. I hope my answer can still help you.

As [doc string](https://docs.rs/handlebars/latest/handlebars/struct.Handlebars.html#method.set_dev_mode) of `set_dev_mode` suggests, you need to enable dev_mode before registering templates.
